receivers/jira/jira.go

```diff
@@ -0,0 +1,376 @@
+	if err != nil {
+		return shouldRetry, fmt.Errorf("failed to look up existing issues: %w", err)
+	}
+
+	var method string
+	var path string
+	if existingIssue == nil {
+		// Do not create new issues for resolved alerts
+		if alerts.Status() == model.AlertResolved {
+			return false, nil
+		}
+		path = "issue"
+		method = http.MethodPost
+		logger.Debug("create new issue")
+	} else {
+		path = "issue/" + url.PathEscape(existingIssue.Key)
+		method = http.MethodPut
+		logger.Debug("updating existing issue", "issue_key", existingIssue.Key)
+	}
+
+	requestBody := n.prepareIssueRequestBody(ctx, logger, key.Hash(), as...)
+
+	_, shouldRetry, err = n.doAPIRequest(ctx, method, path, requestBody)
+	if err != nil {
+		return shouldRetry, fmt.Errorf("failed to %s request to %q: %w", method, path, err)
+	}
+
+	return n.transitionIssue(ctx, logger, existingIssue, alerts.HasFiring())
+}
+
+func (n *Notifier) prepareIssueRequestBody(ctx context.Context, logger logging.Logger, groupID string, as ...*types.Alert) issue {
+	var tmplErr error
+	tmpl, _ := templates.TmplText(ctx, n.tmpl, as, n.log, &tmplErr)
+
+	renderOrDefault := func(fieldName, template, fallback string) string {
+		defer func() {
+			tmplErr = nil
+		}()
+		result := tmpl(template)
+		if tmplErr == nil {
+			return result
+		}
+		if fallback == "" || fallback == template {
+			logger.Error("failed to render template", "error", tmplErr, "configField", fieldName, "template", template)
+			return ""
+		}
+		logger.Error("failed to render template, use default template", "error", tmplErr, "configField", fieldName, "template", template)
+		tmplErr = nil
+		result = tmpl(fallback)
+		if tmplErr == nil {
+			return result
+		}
+		logger.Error("failed to render default template", "error", tmplErr, "configField", fieldName, "template", fallback)
+		return ""
+	}
+
+	summary := renderOrDefault("summary", n.conf.Summary, DefaultSummary)
+	summary, truncated := notify.TruncateInRunes(summary, MaxSummaryLenRunes)
+	if truncated {
+		logger.Warn("Truncated summary", "max_runes", MaxSummaryLenRunes)
+	}
+
+	fields := &issueFields{
+		Summary: summary,
+		Labels:  make([]string, 0, len(n.conf.Labels)+1),
+		Fields:  n.conf.Fields,
+	}
+
+	issueDescriptionString := renderOrDefault("description", n.conf.Description, DefaultDescription)
+	fields.Description = n.prepareDescription(issueDescriptionString, logger)
+
+	projectKey := strings.TrimSpace(renderOrDefault("project", n.conf.Project, ""))
+	if projectKey != "" {
+		fields.Project = &keyValue{Key: projectKey}
+	}
+	issueType := strings.TrimSpace(renderOrDefault("issue_type", n.conf.IssueType, ""))
+	if issueType != "" {
+		fields.Issuetype = &idNameValue{Name: issueType}
+	}
+
+	for i, label := range n.conf.Labels {
+		label = strings.TrimSpace(renderOrDefault(fmt.Sprintf("labels[%d]", i), label, ""))
+		if label == "" {
+			continue
+		}
+		fields.Labels = append(fields.Labels, label)
+	}
+	sort.Strings(fields.Labels)
```

Comment:
Nit: From the [docs](https://pkg.go.dev/sort#Strings), calling `sort.Strings()` in Go 1.22 or higher simply calls `slices.Sort()`.
```suggestion
	slices.Sort(fields.Labels)
```